Sources/Support/POSIX/ELFSupport.cpp

```diff
@@ -69,5 +71,93 @@ bool ELFSupport::MachineTypeToCPUType(uint32_t machineType, bool is64Bit,
   }
   return true;
 }
+
+bool ELFSupport::GetELFFileBuildID(std::string const &path, ByteVector &buildId) {
+  int fd = ::open(path.c_str(), O_RDONLY);
+  if (fd < 0)
+    return false;
+
+  Elf32_Ehdr e32hdr;
+  if (::pread(fd, &e32hdr, sizeof(e32hdr), 0) != sizeof(e32hdr) ||
+      e32hdr.e_ident[EI_MAG0] != ELFMAG0 || e32hdr.e_ident[EI_MAG1] != ELFMAG1 ||
+      e32hdr.e_ident[EI_MAG2] != ELFMAG2 || e32hdr.e_ident[EI_MAG3] != ELFMAG3) {
```

Comment:
Why not `memcmp` it?

Reply:
I'm happy to change to `memcmp` if you prefer. That's how I wrote it initially, but I found two other places in the ds2 source that do it this way and figured I should be consistent. I find it strange that someone bothered to define these constants in `elf.h` to being with...

Reply:
Consistency is good; I think that replacing them as a follow up is reasonable. The indexing shouldn't matter, and the `memcmp` should fold away as it is fixed size and a 32-bit compare should be commonly available on all platforms at this point.